packages/pagination/__tests__/pagination.spec.ts

```diff
@@ -1,407 +1,377 @@
+      const wrapper = mount({
+        setup() {
+          return () => {
+            return h(Pagination, {
+              total: 100,
+            })
+          }
+        },
+      })
+      assertElementsExistence(wrapper, [
+        '.el-pagination__rightwrapper',
+        'button.btn-prev',
+        'ul.el-pager',
+        'button.btn-next',
+        '.el-pagination__jump',
+      ], true)
     })
-    wrapper.find('.btn-prev').trigger('click')
-    await sleep(TIME_OUT)
-    expect(wrapper.vm.internalCurrentPage).toEqual(1)
-    wrapper.find('.btn-prev').trigger('click')
-    expect(wrapper.vm.internalCurrentPage).toEqual(1)
-  })
 
-  test('currentPage', () => {
-    const wrapper = mount(Pagination, {
-      props: {
-        pageSize: 20,
-        total: 200,
-        currentPage: 3,
-      },
+    test('test layout with slot', () => {
+      const TestComponent = {
+        template: `
+          <el-pagination
+            layout="slot, prev, pager, next"
+            :page-size="25"
+            :total="100">
+            <span class="slot-test">slot test</span>
+          </el-pagination>
+        `,
+        components: {
+          'el-pagination': Pagination,
+        },
+      }
+      const wrapper = mount(TestComponent)
+      expect(wrapper.find('.slot-test').exists()).toBe(true)
     })
-    expect(wrapper.find('li.number.active').text()).toEqual('3')
-  })
 
-  test('currentPage: NaN', () => {
-    const wrapper = mount(Pagination, {
-      props: {
-        pageSize: 20,
-        total: 200,
-        currentPage: NaN,
-      },
+    test('test small layout', () => {
+      const wrapper = mount({
+        setup() {
+          return () => {
+            return h(Pagination, {
+              total: 100,
+              small: true,
+            })
+          }
+        },
+      })
+      expect(wrapper.vm.$el.classList.contains('el-pagination--small')).toBe(true)
     })
-    expect(wrapper.find('li.number.active').text()).toEqual('1')
-    expect(wrapper.vm.$el.querySelectorAll('li.number').length).toBe(7)
-  })
 
-  test('layout is empty', () => {
-    const wrapper = mount(Pagination, {
-      props: {
-        layout: '',
-      },
+    test('test with background', async () => {
+      const withBackground = ref(true)
+      const wrapper = mount({
+        setup() {
+          return () => {
+            return h(Pagination, {
+              total: 100,
+              background: withBackground.value,
+            })
+          }
+        },
+      })
+      expect(wrapper.find('.is-background').exists()).toBe(true)
+      withBackground.value = false
+      await nextTick()
+      expect(wrapper.find('.is-background').exists()).toBe(false)
     })
-    expect(wrapper.vm.$el.textContent).toEqual('')
-  })
-})
 
-describe('click pager', () => {
-  test('click ul', () => {
-    const wrapper = mount(Pagination, {
-      props: {
-        total: 1000,
-      },
+    test('test hide-on-single-page prop', async () => {
```

Comment:
simply put `hide-on-single-page` here should be ok, but it is not a mandatory change request here so you can decide yourself whether you want to change it or not.

Reply:
keep word `test` for naming consistency